stage/Game.tsx

```diff
@@ -1,28 +1,109 @@
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect, useRef, useMemo } from "react";
 import Board from "../components/Board";
 import boardWithRandomlyPlacedShips from "../functions/boardWithRandomlyPlacedShips ";
 import { shipLength } from "../interfaces/types";
+import { motion } from "framer-motion";
 
 //Conditions 'Sunk' | 'Hit' | 'Miss' | 'Ship' | 'Empty'
 
 const Game = () => {
   const width = 10;
   const [opponentBoard, setOpponentBoard] = useState<number[][]>(newBoard());
   const [playerBoard, setPlayerBoard] = useState<number[][]>(newBoard());
+  const [orientation, setOrientation] = useState("vertical");
+  const [isShipSelected, setIsShipSelected] = useState(false);
+  const [currentGridXLocation, setCurrentGridXLocation] = useState<number>();
```

Comment:
Would `-1` as default value work? - having to consider its type as `undefined` as well sounds annoying.

Reply:
Good idea. Will try

Reply:
> 
"_Having 100 event handlers in one board is alot_"

It is true that it sounds daunting but I chose this way in hopes that responsive designs can come in to play easily in the future, since it depends entirely on the `div`s